api/jobs/jobs.py

```diff
@@ -78,6 +87,8 @@ def __init__(self, name, inputs, destination=None, tags=None, attempt=1, previou
         self.id_             = id_
         self.config          = config_
         self.now             = now
+        self.origin          = origin
+        self.outputs         = []
```

Comment:
I hate to do this to you, but I think `outputs` should be renamed before this lands - `inputs` and `outputs` (currently a singleton `destination` but not forever) are kinda reserved meaning. I'd take `saved` or `result-filenames` instead?